From a06433e8ccc5033c6c0f7c5d2dfb49072fba9702 Mon Sep 17 00:00:00 2001
From: Shamim Ali <ashamim@ami.com>
Date: Mon, 25 Nov 2024 19:08:59 +0530
Subject: [PATCH] Removed PrivateKey Validation

This patch will remove the private key validation,
currently https and trust store the same 'Install' method
to update the  'trust' and 'https' certificates, since the
private key is not not required for trust certificates, hence
removing this validation, 'Https' type certificate part need
to be handle separately.

Signed-off-by: Shamim Ali <ashamim@ami.com>
---
 certificate.cpp | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/certificate.cpp b/certificate.cpp
index 9069f2c..34370b6 100644
--- a/certificate.cpp
+++ b/certificate.cpp
@@ -347,14 +347,6 @@ void Certificate::install(const std::string& certSrcFilePath, bool restore)
         elog<InternalFailure>();
     }
 
-    validatePrivateKeyEncryption(certSrcFilePath);
-
-    if (!compareKeys(certSrcFilePath))
-    {
-            elog<PrivateKeyCertificateFileNotMatch>(PrivateKeyCertificateFileNotMatchReason(
-                "Private key does not match the Certificate"));
-    }
-
     X509StorePtr x509Store = getX509Store(certSrcFilePath);
 
     // Load Certificate file into the X509 structure.
-- 
2.34.1

